templates/spider_zacatrus/settings.txt
# Scrapy settings for zacatrus_project project
#
# For simplicity, this file contains only settings considered important or
# commonly used. You can find more settings consulting the documentation:
#
#     https://docs.scrapy.org/en/latest/topics/settings.html
#     https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#     https://docs.scrapy.org/en/latest/topics/spider-middleware.html

BOT_NAME = "zacatrus_project"

SPIDER_MODULES = ["zacatrus_project.spiders"]
NEWSPIDER_MODULE = "zacatrus_project.spiders"

# User agent y configuración básica
USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36'
ROBOTSTXT_OBEY = False
CONCURRENT_REQUESTS = 6
CONCURRENT_REQUESTS_PER_DOMAIN = 12


# FEEDS=   {
#             'data/zacatrus.csv': {
#                 'format': 'csv',
#                 'encoding': 'utf8',
#                 'overwrite': True
#             }
#         }

##############################################
###################PLAYWRIGHT#################
##############################################

# SCRAPY_PLAYWRIGHT_ENABLED = True

# DOWNLOAD_HANDLERS = {
#     "http": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
#     "https": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
# }

# DOWNLOADER_MIDDLEWARES = {
#     "scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware": 810,
# }

TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"

# # Configuración de Playwright
# PLAYWRIGHT_BROWSER_TYPE = "chromium"
# PLAYWRIGHT_LAUNCH_OPTIONS = {
#     "headless": False,  # Cambia a True si prefieres no ver el navegador
#     "args": ["--disable-blink-features=AutomationControlled"],
# }

# # Tiempo de espera de navegación (en milisegundos)
# PLAYWRIGHT_DEFAULT_NAVIGATION_TIMEOUT = 60000  # 60 segundos

# # Limites de contextos y páginas (ajusta según tus recursos)
# PLAYWRIGHT_MAX_CONTEXTS = 12
# PLAYWRIGHT_MAX_PAGES_PER_CONTEXT = 6

# Codificación de exportación
FEED_EXPORT_ENCODING = "utf-8"


# Configure a delay for requests for the same website (default: 0)
# See https://docs.scrapy.org/en/latest/topics/settings.html#download-delay
# See also autothrottle settings and docs
#DOWNLOAD_DELAY = 3
# The download delay setting will honor only one of:
#CONCURRENT_REQUESTS_PER_DOMAIN = 16
#CONCURRENT_REQUESTS_PER_IP = 16

# Disable cookies (enabled by default)
#COOKIES_ENABLED = False

# Disable Telnet Console (enabled by default)
#TELNETCONSOLE_ENABLED = False

# Override the default request headers:
#DEFAULT_REQUEST_HEADERS = {
#    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
#    "Accept-Language": "en",
#}

# Enable or disable spider middlewares
# See https://docs.scrapy.org/en/latest/topics/spider-middleware.html
#SPIDER_MIDDLEWARES = {
#    "zacatrus_project.middlewares.ZacatrusProjectSpiderMiddleware": 543,
#}

# Enable or disable downloader middlewares
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#DOWNLOADER_MIDDLEWARES = {
#    "zacatrus_project.middlewares.ZacatrusProjectDownloaderMiddleware": 543,
#}

# Enable or disable extensions
# See https://docs.scrapy.org/en/latest/topics/extensions.html
#EXTENSIONS = {
#    "scrapy.extensions.telnet.TelnetConsole": None,
#}

# Configure item pipelines
# See https://docs.scrapy.org/en/latest/topics/item-pipeline.html
#ITEM_PIPELINES = {
#    "zacatrus_project.pipelines.ZacatrusProjectPipeline": 300,
#}

# Enable and configure the AutoThrottle extension (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/autothrottle.html
#AUTOTHROTTLE_ENABLED = True
# The initial download delay
#AUTOTHROTTLE_START_DELAY = 5
# The maximum download delay to be set in case of high latencies
#AUTOTHROTTLE_MAX_DELAY = 60
# The average number of requests Scrapy should be sending in parallel to
# each remote server
#AUTOTHROTTLE_TARGET_CONCURRENCY = 1.0
# Enable showing throttling stats for every response received:
#AUTOTHROTTLE_DEBUG = False

# Enable and configure HTTP caching (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#httpcache-middleware-settings
#HTTPCACHE_ENABLED = True
#HTTPCACHE_EXPIRATION_SECS = 0
#HTTPCACHE_DIR = "httpcache"
#HTTPCACHE_IGNORE_HTTP_CODES = []
#HTTPCACHE_STORAGE = "scrapy.extensions.httpcache.FilesystemCacheStorage"

# Set settings whose default value is deprecated to a future-proof value
# TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
# FEED_EXPORT_ENCODING = "utf-8"


templates/spider_housinganywhere/settings.txt
import os
from dotenv import load_dotenv

load_dotenv()

# Scrapy settings for housinganywhere_project project
#
# For simplicity, this file contains only settings considered important or
# commonly used. You can find more settings consulting the documentation:
#
#     https://docs.scrapy.org/en/latest/topics/settings.html
#     https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#     https://docs.scrapy.org/en/latest/topics/spider-middleware.html

BOT_NAME = "housinganywhere_project"

SPIDER_MODULES = ["housinganywhere_project.spiders"]
NEWSPIDER_MODULE = "housinganywhere_project.spiders"

SCRAPEOPS_API_KEY = os.getenv("SCRAPEOPS_API_KEY")
SCRAPEOPS_FAKE_USER_AGENT_ENDPOINT = 'https://headers.scrapeops.io/v1/user-agents'
SCRAPEOPS_FAKE_USER_AGENT_ENABLED = True
SCRAPEOPS_NUM_RESULTS = 50

LOG_LEVEL = "INFO"


# User agent y configuración básica
USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36'
ROBOTSTXT_OBEY = False
CONCURRENT_REQUESTS = 4
# CONCURRENT_REQUESTS_PER_DOMAIN = 12

##############################################
###################PLAYWRIGHT#################
##############################################

SCRAPY_PLAYWRIGHT_ENABLED = True

DOWNLOAD_HANDLERS = {
    "http": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
    "https": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
}

TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"

# Configuración de Playwright
PLAYWRIGHT_BROWSER_TYPE = "chromium"
# Máximo de contextos (browsers) simultáneos
# PLAYWRIGHT_MAX_CONTEXTS = 6

# Máximo de páginas (tabs) por contexto
PLAYWRIGHT_MAX_PAGES_PER_CONTEXT = 4

# Lanzamiento visible de los navegadores
PLAYWRIGHT_LAUNCH_OPTIONS = {
    "headless": False,
    "args": ["--disable-blink-features=AutomationControlled"],
}

# Tiempo de espera de navegación (en milisegundos)
PLAYWRIGHT_DEFAULT_NAVIGATION_TIMEOUT = 60000  # 60 segundos

# # Limites de contextos y páginas (ajusta según tus recursos)
# PLAYWRIGHT_MAX_CONTEXTS = 12
# PLAYWRIGHT_MAX_PAGES_PER_CONTEXT = 6

# Codificación de exportación
FEED_EXPORT_ENCODING = "utf-8"


# Configure a delay for requests for the same website (default: 0)
# See https://docs.scrapy.org/en/latest/topics/settings.html#download-delay
# See also autothrottle settings and docs
#DOWNLOAD_DELAY = 3
# The download delay setting will honor only one of:
#CONCURRENT_REQUESTS_PER_DOMAIN = 16
#CONCURRENT_REQUESTS_PER_IP = 16

# Disable cookies (enabled by default)
#COOKIES_ENABLED = False

# Disable Telnet Console (enabled by default)
#TELNETCONSOLE_ENABLED = False

# Override the default request headers:
#DEFAULT_REQUEST_HEADERS = {
#    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
#    "Accept-Language": "en",
#}

# Enable or disable spider middlewares
# See https://docs.scrapy.org/en/latest/topics/spider-middleware.html
SPIDER_MIDDLEWARES = {
#    "housinganywhere_project.middlewares.HousinganywhereProjectSpiderMiddleware": 543,
#    "housinganywhere_project.middlewares.ScrapeOpsFakeBrowserHeaderAgentMiddleware": 400,
}

# Enable or disable downloader middlewares
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
DOWNLOADER_MIDDLEWARES = {
    # Desactiva el middleware de User-Agent por defecto de Scrapy
    "scrapy.downloadermiddlewares.useragent.UserAgentMiddleware": None,

    # Tus middlewares personalizados (AJUSTA PRIORIDADES SI QUIERES)
    "housinganywhere_project.middlewares.ScrapeOpsFakeUserAgentMiddleware": 400,
    # "housinganywhere_project.middlewares.ScrapeOpsFakeBrowserHeaderAgentMiddleware": 410,

    # El de compresión lo dejas como lo tenías
    "scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware": 810,
}
 #es una extensión (middleware de descarga)

# Enable or disable extensions
# See https://docs.scrapy.org/en/latest/topics/extensions.html
#EXTENSIONS = {
#    "scrapy.extensions.telnet.TelnetConsole": None,
#}

# Configure item pipelines
# See https://docs.scrapy.org/en/latest/topics/item-pipeline.html
ITEM_PIPELINES = {
   "housinganywhere_project.pipelines.HousinganywhereProjectPipeline": 300,
}

# Enable and configure the AutoThrottle extension (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/autothrottle.html
#AUTOTHROTTLE_ENABLED = True
# The initial download delay
#AUTOTHROTTLE_START_DELAY = 5
# The maximum download delay to be set in case of high latencies
#AUTOTHROTTLE_MAX_DELAY = 60
# The average number of requests Scrapy should be sending in parallel to
# each remote server
#AUTOTHROTTLE_TARGET_CONCURRENCY = 1.0
# Enable showing throttling stats for every response received:
#AUTOTHROTTLE_DEBUG = False

# Enable and configure HTTP caching (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#httpcache-middleware-settings
#HTTPCACHE_ENABLED = True
#HTTPCACHE_EXPIRATION_SECS = 0
#HTTPCACHE_DIR = "httpcache"
#HTTPCACHE_IGNORE_HTTP_CODES = []
#HTTPCACHE_STORAGE = "scrapy.extensions.httpcache.FilesystemCacheStorage"

# # Set settings whose default value is deprecated to a future-proof value
# TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
# FEED_EXPORT_ENCODING = "utf-8"


src/projects/zacatrus_project/zacatrus_project/settings.py
# Scrapy settings for zacatrus_project project
#
# For simplicity, this file contains only settings considered important or
# commonly used. You can find more settings consulting the documentation:
#
#     https://docs.scrapy.org/en/latest/topics/settings.html
#     https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#     https://docs.scrapy.org/en/latest/topics/spider-middleware.html

BOT_NAME = "zacatrus_project"

SPIDER_MODULES = ["zacatrus_project.spiders"]
NEWSPIDER_MODULE = "zacatrus_project.spiders"

# User agent y configuración básica
USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36'
ROBOTSTXT_OBEY = False
CONCURRENT_REQUESTS = 6
CONCURRENT_REQUESTS_PER_DOMAIN = 12


# FEEDS=   {
#             'data/zacatrus.csv': {
#                 'format': 'csv',
#                 'encoding': 'utf8',
#                 'overwrite': True
#             }
#         }

##############################################
###################PLAYWRIGHT#################
##############################################

# SCRAPY_PLAYWRIGHT_ENABLED = True

# DOWNLOAD_HANDLERS = {
#     "http": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
#     "https": "scrapy_playwright.handler.ScrapyPlaywrightDownloadHandler",
# }

# DOWNLOADER_MIDDLEWARES = {
#     "scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware": 810,
# }

TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"

# # Configuración de Playwright
# PLAYWRIGHT_BROWSER_TYPE = "chromium"
# PLAYWRIGHT_LAUNCH_OPTIONS = {
#     "headless": False,  # Cambia a True si prefieres no ver el navegador
#     "args": ["--disable-blink-features=AutomationControlled"],
# }

# # Tiempo de espera de navegación (en milisegundos)
# PLAYWRIGHT_DEFAULT_NAVIGATION_TIMEOUT = 60000  # 60 segundos

# # Limites de contextos y páginas (ajusta según tus recursos)
# PLAYWRIGHT_MAX_CONTEXTS = 12
# PLAYWRIGHT_MAX_PAGES_PER_CONTEXT = 6

# Codificación de exportación
FEED_EXPORT_ENCODING = "utf-8"


# Configure a delay for requests for the same website (default: 0)
# See https://docs.scrapy.org/en/latest/topics/settings.html#download-delay
# See also autothrottle settings and docs
#DOWNLOAD_DELAY = 3
# The download delay setting will honor only one of:
#CONCURRENT_REQUESTS_PER_DOMAIN = 16
#CONCURRENT_REQUESTS_PER_IP = 16

# Disable cookies (enabled by default)
#COOKIES_ENABLED = False

# Disable Telnet Console (enabled by default)
#TELNETCONSOLE_ENABLED = False

# Override the default request headers:
#DEFAULT_REQUEST_HEADERS = {
#    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
#    "Accept-Language": "en",
#}

# Enable or disable spider middlewares
# See https://docs.scrapy.org/en/latest/topics/spider-middleware.html
#SPIDER_MIDDLEWARES = {
#    "zacatrus_project.middlewares.ZacatrusProjectSpiderMiddleware": 543,
#}

# Enable or disable downloader middlewares
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
#DOWNLOADER_MIDDLEWARES = {
#    "zacatrus_project.middlewares.ZacatrusProjectDownloaderMiddleware": 543,
#}

# Enable or disable extensions
# See https://docs.scrapy.org/en/latest/topics/extensions.html
#EXTENSIONS = {
#    "scrapy.extensions.telnet.TelnetConsole": None,
#}

# Configure item pipelines
# See https://docs.scrapy.org/en/latest/topics/item-pipeline.html
#ITEM_PIPELINES = {
#    "zacatrus_project.pipelines.ZacatrusProjectPipeline": 300,
#}

# Enable and configure the AutoThrottle extension (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/autothrottle.html
#AUTOTHROTTLE_ENABLED = True
# The initial download delay
#AUTOTHROTTLE_START_DELAY = 5
# The maximum download delay to be set in case of high latencies
#AUTOTHROTTLE_MAX_DELAY = 60
# The average number of requests Scrapy should be sending in parallel to
# each remote server
#AUTOTHROTTLE_TARGET_CONCURRENCY = 1.0
# Enable showing throttling stats for every response received:
#AUTOTHROTTLE_DEBUG = False

# Enable and configure HTTP caching (disabled by default)
# See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html#httpcache-middleware-settings
#HTTPCACHE_ENABLED = True
#HTTPCACHE_EXPIRATION_SECS = 0
#HTTPCACHE_DIR = "httpcache"
#HTTPCACHE_IGNORE_HTTP_CODES = []
#HTTPCACHE_STORAGE = "scrapy.extensions.httpcache.FilesystemCacheStorage"

# Set settings whose default value is deprecated to a future-proof value
# TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
# FEED_EXPORT_ENCODING = "utf-8"
